tutorials/foreign-object.html

```diff
@@ -0,0 +1,813 @@
+            <li><a href="https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button" target="_blank"><code>button</code></a></li>
+            <li><a href="https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select" target="_blank"><code>select</code></a></li>
+            <li><a href="https://developer.mozilla.org/en-US/docs/Web/HTML/Element/option" target="_blank"><code>option</code></a></li>
+        </ul>
+
+        <p>
+            In the example above, we created a paper event with the aim of managing focus. One detail you may not have noticed is that the 
+            <code>button</code> text is actually contained within a <code>span</code> element. If we had not applied the <code>paper</code> event, 
+            and the <code>input</code> was in focus, clicking on the <code>span</code> element won't actually cause the input to lose focus. However, 
+            clicking on the <code>button</code> itself will make the <code>input</code> lose focus, as <code>button</code> is one of the 
+            elements where JointJS prevents the default interaction.
+        </p>
+
+        <p>
+            If we wanted the <code>input</code> to lose focus when a user clicks on the <code>span</code> element, we can take advantage of the
+            <code>guard</code> option as follows <code>guard: (evt) => ['SPAN'].includes(evt.target.tagName)</code>. Although, including the 
+            <code>guard</code> option to manage focus in this instance would be superfluous if the <code>paper</code> event is already applied.
+        </p>
+
+        <p>
+            One of the final items you may wish to take care of when using a <code>form</code> in JointJS is to prevent the default browser behaviour 
+            when submitting user data. In order to achieve this, we will create a custom 
+            <a href="/docs/jointjs#dia.CellView.custom" target="_blank"><code>CellView</code></a> for our element. It will contain an events hash 
+            which specifies a DOM event that will be bound to a method on the view. The <i>key</i> is made up of the event name plus a CSS selector, 
+            and the <i>value</i> is a method on our custom view.
+        </p>
+
+        <p>
+            In the following example, we specify the event name <code>'submit'</code>, plus our CSS element selector <code>'form'</code>, and lastly 
+            the name of our method <code>'onSubmit'</code>. We can prevent that browser refresh by using <code>evt.preventDefault()</code>, and clear 
+            the value of the HTML input value afterwards if we wish to do so.
+        </p>
+
+        <pre><code>joint.shapes.example.FormView = joint.dia.ElementView.extend({
+
+    events: {
+        'submit form': 'onSubmit'
+    },
+
+    onSubmit: function(evt) {
+        evt.preventDefault();
+        evt.target.children.name.value = '';
+        this.model.attr('name/props/value', '');
+    },
+
+});
+</code></pre>
+
+        <h4>Props, a special attribute</h4>
```

Comment:
is this supposed to be h4 - i.e. a subheading of "A more interactive example" section? I think it can be a separate h3 section